### DISCUSSION TITLE
Error executing verb "cluster_graph" in create_base_entity_graph: Columns must be same length as key

### DISCUSSION THREAD

**Discussion by subho-das** [microsoft/graphrag#653] [Q&A] [answered]
Getting this error when running graphrag. 

`09:24:52,917 datashaper.workflow.workflow INFO executing verb cluster_graph
09:24:52,918 graphrag.index.verbs.graph.clustering.cluster_graph WARNING Graph has no nodes
09:24:52,922 datashaper.workflow.workflow ERROR Error executing verb "cluster_graph" in create_base_entity_graph: Columns must be same length as key
Traceback (most recent call last):`

The indexing code is out of the box and I have not made any changes. Couln't figure out how can the column lengths be different?

If anyone has been through this issue , can you please advise?

**Comment by etiennebonnafoux** [comment]
look at #631 #514 
It seems that it is a problem with the output of the LLM, but the fix is unclear

**Comment by awaescher** [comment, marked as answer]
One reason might be that the LLM does not format its responses as expected by GraphRAG.
[A fix that worked for me was to change the LLM model.](https://github.com/microsoft/graphrag/issues/443#issuecomment-2248000519)

**Reply by subho-das** [reply]
I see. Yes the cluster_graph issue was resolved by using the mistral model. However I could not find an easy way to visualize the graph. The nodes always came without labels. Can you please advise some easy way to understand the entities extracted from a document that I fed into graphrag and how it was processed into community level hierarchies ? Entity relationship along with the hierarchy view is the thing I am looking for.

**Reply by awaescher** [reply]
I inspected my graphml files with Gephi which does not show labels by default. I found out that I had to activate them with this button on the bottom toolbar:

![image](https://github.com/user-attachments/assets/75d3d91f-e22c-463a-a4ab-7a7ef3f84723)

However, these graphs were not very useful to me, I am not sure yet whether Gephi did a poor job rendering them or if the graphs itself contain not enough information.
